commands/snapshotautocreatecommand.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * @file
+ * @version 0.1
+ * @copyright 2017 CN-Consult GmbH
+ * @author Jens Stahl <jens.stahl@cn-consult.eu>
+ *
+ * License: Please check the LICENSE file for more information.
+ */
+
+namespace PBST\Command;
+
+use ProfitBricksApi\Snapshot;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use ProfitBricksApi\ProfitBricksApi;
+use ProfitBricksApi\DataCenter;
+use DateTime;
+use DateInterval;
+use Exception;
+
+/**
+ * Class SnapshotAutoCreateCommand
+ *
+ * Creates automated snapshots from virtual servers which have been configured in config.ini.
+ */
+class SnapshotAutoCreateCommand extends Command
+{
+    private $config;
+    private $virtualMachineState;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->virtualMachineState = array();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName("snapshot:autoCreate")
+            ->setDescription("Creates a snapshot from all disks which are attached to a server when necessary.");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (is_readable('config.ini'))
+        {
+            $this->config = parse_ini_file('config.ini', true);
+            if (!isset($this->config['api']['user']) || !isset($this->config['api']['password']))throw new Exception("No user or no password configured to connect ProfitBricks!");
```

Comment:
This code could run into PHP7-problems with the variable handling order changes.
If it comes to invalid array-to-string-conversions, there have to be added some curly braces:

for example:` isset($this->{config['api']['user']})`  reverts the code to the intended behaviour